src/core/net/nd6.hpp

```diff
@@ -557,8 +652,11 @@ class RouterAdvertMessage
         //  |   Options ...
         //  +-+-+-+-+-+-+-+-+-+-+-+-
 
-        static constexpr uint8_t kPreferenceOffset = 3;
-        static constexpr uint8_t kPreferenceMask   = 3 << kPreferenceOffset;
+        static constexpr uint8_t kManagedAddressFlag = 1 << 7;
+        static constexpr uint8_t kOtherConfigFlag    = 1 << 6;
+        static constexpr uint8_t kHomeAgentFlag      = 1 << 5;
```

Comment:
BTW, my assumption about order here is that bit 0 which is `M` (from RFC) would be `1 << 7` (which seems to match other cases)?

Reply:
Yes, I believe that is correct.

Reply:
Question: the IANA registry for the flags (https://www.iana.org/assignments/icmpv6-parameters/icmpv6-parameters.xhtml#icmpv6-parameters-11) lists one additional flag ( 1 << 2 )
Do we want to add that one as well? (the RA message diagram doesn't list it)

Reply:
I prefer the model where we only add constants and definitions in the code when we need and use them, following the "add code when we use it" guideline. I must admit I may have not myself always followed this rule in the past, but I think it's a good practice. With that in mind, I think I will remove  `kHomeAgentFlag` in the next push. Thoughts?

Reply:
Ok to remove it, if not used! In the code comment that draws the packet layout with the flags, I would also avoid marking bits as 'Reserved' in this case because multiple IETF sources give multiple definitions of whether these are reserved are actually in use.  We may just use dots (...) to indicate there is something there without flag details.